### PR TITLE
Extract some javascript from main Admin and base script

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this can only be done on master.

Part of #7049

## Changelog

```markdown
### Changed
- Refactor internal Sonata Assets to split them in smaller components
```

If we like this idea, I can refactor in small PRs all the current Javascript (and I am sure I can do it for 4.0 release, in time)

### CODE REVIEW

**Comment by jordisala1991** [comment]
WDYT about this kind of refactor @sonata-project/contributors . My idea is that it does not affect the Admin object that is accesible outside the scripts, so it won't break BC.

The thing is. IMO we should provide a way for people to build its own sonata js, to do so with some guarantees, we should provide some scripts that are kind of "public", but have some other that are internal.

Maybe we should provide:

`app.js` in case you want default scripts and want to add some more thing
`admin.js` in case you want default sonata scripts and add your own third party dependencies (like if you want to upgrade to admin-lte 3 in your own project)

And the rest we could treat them as internal OR expose them too. IMO since the files are not all well refactored, I am more of the opinion of internal, but at some point, with good refactored code, we might mark them as all public and live with it...

**Example:**

I want to build my own js and I pick all dependencies from my own node_modules because I want to upgrade admin-lte and I pick most of the sonata js code, BUT I want to override the getConfig function, because I want to pick config from somewhere else. 
I do not know if that makes sense at all, but let's talk about it.

File: assets/js/admin-log.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ */
+const log = (...args) => {
+  if (!getConfig('DEBUG')) {
+    return;
+  }
+
+  const message = `[Sonata.Admin] ${args.join(', ')}`;
+
+  // eslint-disable-next-line no-console
+  console.log(message);
```
Comment (VincentLanglet) [inline assets/js/admin-log.js:24]:
```suggestion
  console.debug(message);
```
?

Comment (jordisala1991) [reply assets/js/admin-log.js]: Changed

**Comment by jordisala1991** [comment]
Please review @sonata-project/contributors . I need your opinions to move this forward or not. This is needed if we want to give to create some documentation on how to customize the assets, and in order to integrate the inline scripts

**Review by VincentLanglet — APPROVED** [review]

**Comment by jordisala1991** [comment]
I want to pass this as draft for now. Maybe there are other important things to do on Sonata before taking this one.

**Pull request closed without merging**